src/Router.sol

```diff
@@ -232,6 +308,37 @@ contract Router is IRouter, EIP712, FeeGenerator {
         return address(agent);
     }
 
+    /// @notice Allow another address to execute on user's behalf
+    /// @param delegatee The address to be delegated
+    /// @param expiry The expiry of the delegation
+    function allow(address delegatee, uint256 expiry) external {
+        _allow(msg.sender, delegatee, expiry);
+    }
+
+    /// @notice Set delegation information via signature
+    /// @param delegation The delegation information to be signed by a signer using EIP-712
+    /// @param signer The signer address
+    /// @param signature The signer's signature bytes
+    function allowBySig(IParam.Delegation calldata delegation, address signer, bytes calldata signature) external {
+        // Verify deadline, signer and signature
+        uint256 deadline = delegation.deadline;
+        if (block.timestamp > deadline) revert SignatureExpired(deadline);
```

Comment:
I'm thinking if we should check both deadline and nonce for allowBySig as it's critical unlike signer fee which aloows reused signature before deadline. What do you think?

Ref
- [Comet](https://github.com/compound-finance/comet/blob/main/contracts/CometExt.sol#L204) checks both nonce and deadline.
- [Permit2](https://github.com/Uniswap/permit2/blob/main/src/libraries/Allowance.sol#L26) packs nonce, expiry and amount to reduce gas.

Reply:
Indeed. Will do.

Reply:
@Jeff-CCH, Router's inheritance of EIP-712 prevents cross-chain replay.